src/GridFS/Bucket.php

```diff
@@ -301,7 +302,7 @@ public function drop()
      * @see Find::__construct() for supported options
      * @param array|object $filter  Query by which to filter documents
      * @param array        $options Additional options
-     * @return Cursor
+     * @return CursorInterface&Iterator
```

Comment:
The `Collection` instance is created internally, so I don't see how someone could pass on a codec to get a different result. The return type will always be `Cursor`.

Reply:
The `$options` passed to `Bucket::find` are passed down to `Collection::find` as-is, so a user could introduce a codec that way, which would result in a return value of a type other than `Cursor`.

That said, this makes me wonder whether we should directly support a `codec` option in the `Bucket` class as well (similar to how `typeMap` is already supported). I haven't considered that until now, so if we decide to go that route I'd do so in a separate ticket and pull request.

Reply:
> The $options passed to Bucket::find are passed down to Collection::find as-is, so a user could introduce a codec that way, which would result in a return value of a type other than Cursor.

Noted that this explains the `CursorInterface&Iterator` return type being added here.

> That said, this makes me wonder whether we should directly support a codec option in the Bucket class as well (similar to how typeMap is already supported).

The `typeMap` option is only used for `getFileDocumentForStream()` and `getFileIdForStream()`, and `uploadFromStream()` indirectly since that calls `getFileIdForStream()`.

I think it does make sense to support `codec` there as we do `typeMap` today. Please create a separate ticket for that.

Reply:
Created PHPLIB-1220.